src/sas/sascalc/dataloader/readers/red2d_reader.py

```diff
@@ -292,22 +284,18 @@ def get_file_contents(self):
         # optional data: if all of dq data == 0, do not pass to output
         if len(dqx_data) == len(qx_data) and dqx_data.any() != 0:
             # if no dqx_data, do not pass dqy_data.
-            #(1 axis dq is not supported yet).
+            # (1 axis dq is not supported yet).
             if len(dqy_data) == len(qy_data) and dqy_data.any() != 0:
                 # Currently we do not support dq parr, perp.
-                # tranfer the comp. to cartesian coord. for newer version.
+                # transfer the comp. to cartesian coord. for newer version.
                 if ver != 1:
                     diag = np.sqrt(qx_data * qx_data + qy_data * qy_data)
                     cos_th = qx_data / diag
                     sin_th = qy_data / diag
-                    self.current_dataset.dqx_data = np.sqrt((dqx_data * cos_th) * \
-                                                 (dqx_data * cos_th) \
-                                                 + (dqy_data * sin_th) * \
-                                                  (dqy_data * sin_th))
-                    self.current_dataset.dqy_data = np.sqrt((dqx_data * sin_th) * \
-                                                 (dqx_data * sin_th) \
-                                                 + (dqy_data * cos_th) * \
-                                                  (dqy_data * cos_th))
+                    self.current_dataset.dqx_data = np.sqrt(
+                        (dqx_data * cos_th)**2 + (dqy_data * sin_th)**2)
+                    self.current_dataset.dqy_data = np.sqrt(
+                        (dqx_data * sin_th)**2 + (dqy_data * cos_th)**2)
```

Comment:
`np.hypot(a, b)` computes `sqrt(a**2 + b**2)` element-wise. Apparently this is a C99 math function, so it should be pretty common everywhere.